### PR TITLE
Implement Backoff/Retry-After support

### PR DESCRIPTION
**Pull request by eoger**
Fixes [SDK-15](https://jira.mozilla.com/browse/SDK-15).

I based my implementation on the [Kinto docs](https://docs.kinto-storage.org/en/latest/api/1.x/backoff.html?highlight=retry), the code itself is a spin-off of what we do in the a-s fxa-client.

### CODE REVIEW

File: nimbus/src/error.rs [hunk 1]
```diff
@@ -15,6 +15,8 @@ pub enum Error {
     RkvError(rkv::StoreError),
     #[error("IO error: {0}")]
     IOError(#[from] std::io::Error),
+    #[error("Number parsing error: {0}")]
```
Comment (mhammond) [inline nimbus/src/error.rs:18]: I'm not sure it's worth exposing this to consumers - I'd suggest just ignoring the header if it is malformed in this way.

**Review by mhammond — CHANGES_REQUESTED** [review]
Most comments are nits, but I'm not sure the per-path handling is what we want?

File: nimbus/src/client/http_client.rs [hunk 2]
```diff
@@ -33,25 +50,67 @@ impl Client {
...
             Err(Error::ResponseError(resp.text().to_string()))
         }
     }
+
+    fn ensure_no_backoff(&self, request: &Request) -> Result<()> {
+        let path = request.url.path();
+        if let Some(EndpointState::Backoff {
+            observed_at,
+            duration,
+        }) = self.endpoints_state.get(path)
```
[the remark this answers is not in the document]
Comment (eoger) [reply nimbus/src/client/http_client.rs:73]: Looks like you're right for backoff: https://github.com/Kinto/kinto/blob/46047070aa5d0b7fda569e836513d46edbb42754/kinto/core/initialization.py#L135-L153

**Review by mhammond — APPROVED** [review]
Awesome, thanks!

**Pull request merged**